.github/workflows/checks.yml

```diff
@@ -25,7 +25,3 @@ jobs:
         uses: coverallsapp/github-action@v2
         with:
           path-to-lcov: coverage.xml
-      - name: Run dead code analysis
-        run: ./bin/find_dead_code
```

Comment:
Are we going to keep this around since `vulture` isn't in `ruff`?

Reply:
yea, let's keep vulture, I put it into the makefile/pyproject.toml in a later PR